jung-visualization/src/main/java/edu/uci/ics/jung/visualization/spatial/AbstractSpatial.java

```diff
@@ -59,6 +57,35 @@ public void setActive(boolean active) {
     this.active = active;
   }
 
+  protected NT getClosest(Collection<NT> nodes, double x, double y, double radius) {
+
+    // since I am comparing with distance squared, i need to square the radius
+    double radiusSq = radius * radius;
+    if (nodes.size() > 0) {
```

Comment:
style nit: I generally prefer to do early exit in these cases:

```java
if (nodes.isEmpty()) {
  return null;
}
double radiusSq = ...;
```

This both decreases indentation in a lot of the code and makes it easier to read the logic.

Reply:
tough call: 2 extra spaces of indent vs multiple return points

Reply:
I actually prefer to have multiple return points in this context.  That's what I meant by "easier to read the logic".  

'if' statements with big bodies (especially if nested) are problematic because by the time you get to the 'else', you may not even be able to see the initial if condition.  

If there are cheap-to-check exit conditions, I think it's better to do those up front, get them out of the way, and then get to the meat of the method.  It makes it much more clear what the exit conditions actually are.